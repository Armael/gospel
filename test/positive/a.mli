(*@ open Ocamlstdlib *)

type t1
type 'a t2

(*@ function f1 (x: t1) : int t2 *)

(*@ function f2 (x: int) : t1 *)

(*@ function f3 (x: int) : int t2 = f1 (f2 x) *)

(* EXPECTED
   [125] File "a.mli", line 1, characters 0-34:
         Error: No module with name `Ocamlstdlib'.
*)
