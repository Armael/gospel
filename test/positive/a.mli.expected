
*******************************
********** Parsed file ********
*******************************
[@@@gospel {| open Ocamlstdlib |}]
type t1
type 'a t2
[@@@gospel {| function f1 (x: t1) : int t2 |}]
[@@@gospel {| function f2 (x: int) : t1 |}]
[@@@gospel {| function f3 (x: int) : int t2 = f1 (f2 x) |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

(*@ open Ocamlstdlib *)

type t1
  

type 'a t2
  

(*@ function f1 ... *)

(*@ function f2 ... *)

(*@ function f3 ... *)
gospel: internal error, uncaught exception:
        Gospel.Utils.Located(_, _)
        
